app/upgrades/v4rc0/upgrades.go

```diff
@@ -0,0 +1,209 @@
+package v4rc0
+
+import (
+	"context"
+	"errors"
+
+	"cosmossdk.io/collections"
+	corestoretypes "cosmossdk.io/core/store"
+	"cosmossdk.io/math"
+	store "cosmossdk.io/store/types"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/runtime"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/cosmos/cosmos-sdk/types/query"
+	stkkeeper "github.com/cosmos/cosmos-sdk/x/staking/keeper"
+	stktypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	"github.com/babylonlabs-io/babylon/v4/app/keepers"
+	"github.com/babylonlabs-io/babylon/v4/app/upgrades"
+	btcstkkeeper "github.com/babylonlabs-io/babylon/v4/x/btcstaking/keeper"
+	btcstktypes "github.com/babylonlabs-io/babylon/v4/x/btcstaking/types"
+	costktypes "github.com/babylonlabs-io/babylon/v4/x/coostaking/types"
+)
+
+// UpgradeName defines the on-chain upgrade name for the Babylon v4rc0 upgrade
+const UpgradeName = "v4rc0"
+
+// Upgrade for version v4rc0
+var Upgrade = upgrades.Upgrade{
+	UpgradeName:          UpgradeName,
+	CreateUpgradeHandler: CreateUpgradeHandler,
+	StoreUpgrades: store.StoreUpgrades{
+		Added:   []string{costktypes.StoreKey},
+		Deleted: []string{},
+	},
+}
+
+func CreateUpgradeHandler(mm *module.Manager, configurator module.Configurator, keepers *keepers.AppKeepers) upgradetypes.UpgradeHandler {
+	return func(ctx context.Context, plan upgradetypes.Plan, fromVM module.VersionMap) (module.VersionMap, error) {
+		// Run migrations before applying any other state changes.
+		migrations, err := mm.RunMigrations(ctx, configurator, fromVM)
+		if err != nil {
+			return nil, err
+		}
+
+		costkStoreKey := keepers.GetKey(costktypes.StoreKey)
+		if costkStoreKey == nil {
+			return nil, errors.New("invalid coostaking types store key")
+		}
+
+		coStkStoreService := runtime.NewKVStoreService(costkStoreKey)
+		if err := InitializeCoStakerRwdsTracker(ctx, keepers.EncCfg.Codec, coStkStoreService, keepers.StakingKeeper, keepers.BTCStakingKeeper); err != nil {
+			return nil, err
+		}
+
+		return migrations, nil
+	}
+}
+
+// InitializeCoStakerRwdsTracker initializes the coostaker rewards tracker
+// It looks for all BTC stakers that are also baby stakers
+func InitializeCoStakerRwdsTracker(
+	ctx context.Context,
+	cdc codec.BinaryCodec,
+	costkStoreService corestoretypes.KVStoreService,
+	stkKeeper *stkkeeper.Keeper,
+	btcStkKeeper btcstkkeeper.Keeper,
+) error {
+	btcStakers, err := getAllBTCStakers(ctx, btcStkKeeper)
```

Comment:
We could iterate over the data in incentives as well, might be faster than over all btc delegations since incentives already groups the btc stakers by babylon addr

Reply:
But the incentives keeps track of the rewards only, no? There may be the case where there's a user that has pending rewards, but is not actively staking bitcoin.
The getAllBTCStakers returns all active BTC delegators